deps.ffmpeg/60-mbedtls.zsh

```diff
@@ -0,0 +1,172 @@
+autoload -Uz log_debug log_error log_info log_status log_output
+
+name='mbedtls'
+local version='2.24.0'
+local url='https://github.com/ARMmbed/mbedtls.git'
+local hash='523f0554b6cdc7ace5d360885c3f5bbcc73ec0e8'
+local -a patches=(
+  "macos ${0:a:h}/patches/mbedtls/0001-enable-posix-threading-support.patch \
+    363e6b8359f1c5fb8cc8e3c47439223c79de09935697e1d38c20336529fb9a5d"
+  "linux ${0:a:h}/patches/mbedtls/0001-enable-posix-threading-support.patch \
+    363e6b8359f1c5fb8cc8e3c47439223c79de09935697e1d38c20336529fb9a5d"
+  "windows ${0:a:h}/patches/mbedtls/0001-enable-alt-threading-mode.patch \
+    306b8aaee8f291cc0dbd4cbee12ea185e722469eb06b8b7113f0a60feca6bbe6"
+  "windows ${0:a:h}/patches/mbedtls/0002-add-alt-threading-header-file.patch \
+    d0dde0836dc6b100edf218207feffbbf808d04b1d0065082cdc5c838f8a4a7c7"
+)
+
+local -i shared_libs=1
+
+setup() {
+  log_info "Setup (%F{3}${target}%f)"
+  setup_dep ${url} ${hash}
+}
+
+clean() {
+  cd "${dir}"
+
+  if [[ ${clean_build} -gt 0 && -d "build_${arch}" ]] {
+    log_info "Clean build directory (%F{3}${target}%f)"
+
+    rm -rf "build_${arch}"
+  }
+}
+
+patch() {
+  autoload -Uz apply_patch
+
+  log_info "Patch (%F{3}${target}%f)"
+
+  cd "${dir}"
+
+  local patch
+  local _target
+  local _url
+  local _hash
+  for patch (${patches}) {
+    read _target _url _hash <<< "${patch}"
+
+    if [[ ${_target} == "${target%%-*}" ]] apply_patch "${_url}" "${_hash}"
+  }
+}
+
+config() {
+  autoload -Uz mkcd progress
+
+  local _onoff=(OFF ON)
+
+  args=(
+    ${cmake_flags}
+    -DUSE_SHARED_MBEDTLS_LIBRARY="${_onoff[(( shared_libs + 1 ))]}"
+    -DUSE_STATIC_MBEDTLS_LIBRARY=ON
```

Comment:
See above, we always want static builds (so we specify it by default) and disable shared builds based on preference. These values aren't implicitly tied with each other (otherwise it would be a single static/shared switch).

Reply:
Got it. I'll have to check how this works in practice with building the final Windows deps package.